public/app/features/dataframe-import/utils.ts

```diff
@@ -27,24 +26,30 @@ export function formatFileTypes(acceptedFiles: Accept) {
 export function filesToDataframes(files: File[]): Observable<FileImportResult> {
   return new Observable<FileImportResult>((subscriber) => {
     let completedFiles = 0;
-    files.forEach((file) => {
-      const reader = new FileReader();
-      reader.readAsArrayBuffer(file);
-      reader.onload = () => {
-        const result = reader.result;
-        if (result && result instanceof ArrayBuffer) {
-          if (file.type === 'application/json') {
-            const decoder = new TextDecoder('utf-8');
-            const json = JSON.parse(decoder.decode(result));
-            subscriber.next({ dataFrames: [toDataFrame(json)], file: file });
-          } else {
-            subscriber.next({ dataFrames: readSpreadsheet(result), file: file });
-          }
-          if (++completedFiles >= files.length) {
-            subscriber.complete();
-          }
-        }
-      };
-    });
+    import('app/core/utils/sheet')
```

Comment:
This is cool; I've never seen an import used inside a function definition like this before. So, this will then only import/load the sheets mod from `app/core/utils/sheet` when this function is actually invoked, is that correct?

Reply:
That's the idea.

Reply:
we do Monaco loading the same way, see https://github.com/grafana/grafana/issues/69463#issuecomment-1575027061